airbyte-integrations/bases/base-python/CDK-README.md

```diff
@@ -0,0 +1,196 @@
+(as returned by the `parse_response`function mentioned above). This allows sync to resume from where the previous sync last stopped,
+regardless of success or failure. This function typically compares the state object's and the latest record's cursor field, picking the latest one.
+
+This is paired with the `stream_slices` function. Conceptually, a Stream Slice is the smallest unit of data to be
+synced. In the HTTP case, each Slice is equivalent to a HTTP request; indeed the CDK will make one request
+per element returned by the `stream_slices` function. A Slice object is not typed, and the developer
+is free to include any information necessary to make the request. This function is called when the
+`HTTPStream` is first created. Typically, the `stream_slices` function, via inspecting the state object,
+generates a Slice for every request to be made.
+
+As an example, suppose an API is able to dispense data hourly. If the last sync was exactly 24 hours ago,
+we can either make an API call retrieving all data at once, or make 24 calls each retrieving an hour's
+worth of data. In the latter case, the `stream_slices` function, sees that the previous state contains
+yesterday timestamp, and returns a list of 24 Slices, each with a different hourly timestamp to be
+used when creating request.
+
+Lastly, the `request_params` and `path` functions allow the per-request information contained in a Slice
+to be injected into a request. 
+
+In summary, the incremental stream requires:
+* the `cursor_field` property
+* the `get_updated_state` function
+* the `stream_slices` function
+* updating the `request_params` and `path` function to incorporate slices
+
+#### Secondary Features
+
+The CDK offers other features that make writing HTTP APIs a breeze.
+
+##### Authentication
+
+The CDK supports token and OAuth2.0 authentication via the `TokenAuthenticator` and `Oauth2Authenticator` classes
+respectively. Both authentication strategies are identical in that they place the api token in the `Authorization`
+header. The `OAuth2Authenticator` goes an additional step further and has mechanisms to, given a refresh token,
+refresh the current access token. Note that the `OAuth2Authenticator` currently only supports refresh tokens
+and not the full OAuth2.0 loop.
+
+Using either authenticator is as simple as passing the created authenticator into the relevant `HTTPStream`
+constructor. Here is an [example](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-stripe/source_stripe/source.py#L242) from the Stripe API.
+
+##### Pagination
+
+Most APIs, when facing a large call, tend to return the results in pages. The CDK accommodates paging
+via the `next_page_token` function. This function is meant to extract the next page token from the latest
+response. The CDK will continue making requests as long as the `next_page_function` continues returning
```

Comment:
```suggestion
response. The contents of a "token" are completely up to the developer: it can be an ID, a page number, a partial URL etc.. The CDK will continue making requests as long as the `next_page_token` function continues returning
```